### PR TITLE
Feature/fix async transformer syntax

### LINKED ISSUE #59: Regression between 1.8.2 and 2.0.0
When I upgrade to `2.0.0`, I get following error
```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/sandbox/node_modules/svelte-je
ster/src/svelteconfig' imported from /sandbox/node_modules/svelte-jester/src/transformer.js
```
Here's the [sandbox](https://codesandbox.io/s/dreamy-water-bcnmd?file=/tests/button.test.ts:0-105) where this error can be seen.

If I revert `svelte-jester` to `1.8.2`, sandbox gives a different error but the same code works fine in my local system.

### PR DESCRIPTION
**Pull request by sebastianrothe**
I've tried with my current project and that is the way it works for me.

Should fix #59 

Jest's support for ESM is still experimental and some stuff still doesn't work. (see https://github.com/facebook/jest/issues/9430 and https://jestjs.io/docs/ecmascript-modules)

### CODE REVIEW

**Comment by benmccann** [comment]
I have no idea what changed here with all the formatting changes, so lgtm me I guess?

**Comment by mihar-22** [comment]
I guess the changes can be seen in the second commit in this PR [`fix processAsync transformer export`](https://github.com/mihar-22/svelte-jester/pull/60/commits/1f4bb046a904821745edce64a4ef07ec8b89afac). LGTM!

**Comment by bhvngt** [comment]
after applying this fix, I am getting following error

```
 Jest: synchronous transformer /node_modules/.pnpm/svelte-jester@2.0.1/node_modules/svelte-jester/src/transformer.js must export a "process" function.
```

Here's [sandbox](https://codesandbox.io/s/dreamy-water-bcnmd?file=/tests/button.test.ts) where you can see it happening.

File: src/transformer.js [hunk 1]
```diff
@@ -1,58 +1,53 @@
...
-});
+    code: result.js.code,
+    map: JSON.stringify(result.js.map)
+  }
+}
 
-export default createTransformer;
+export default {
+  processAsync
+}
```
Comment (mihar-22) [inline src/transformer.js:53]: Needs to probably change to `export default processAsync` @sebastianrothe? I haven't read the new transformer docs so I'll touch back on this when I have time unless it's solved earlier.

**Comment by sebastianrothe** [comment]
> I have no idea what changed here with all the formatting changes, so lgtm me I guess?

Yeah, I don't know why the old code was not formatted?

File: src/transformer.js (hunk 1 above)
Comment (sebastianrothe) [reply src/transformer.js:53]: I have only found these references:
- https://github.com/sodatea/vite-jest/blob/main/packages/vite-jest/index.js
- https://github.com/axe-me/vite-jest/blob/main/src/transformer.ts
- https://github.com/bitttttten/jest-transformer-mdx/blob/da6eae4b9fe089643636092eeed7dc5f1a370673/mdx-options.cjs

Most of the transformers (like vue-jest, esbuild-jest) still use the old syntax and/or no ESM.

**Comment by sebastianrothe** [comment]
> after applying this fix, I am getting following error
> 
> ```
>  Jest: synchronous transformer /node_modules/.pnpm/svelte-jester@2.0.1/node_modules/svelte-jester/src/transformer.js must export a "process" function.
> ```
> 
> Here's [sandbox](https://codesandbox.io/s/dreamy-water-bcnmd?file=/tests/button.test.ts) where you can see it happening.

For the new code to work, you need to enabled ESM support in Jest. Which means *ALL* steps described here

I think we need to refactor this code a little bit more and provide a timeline for different versions:
- 1.x -> should still be CJS and support Jest 26
- 2.x -> should still be CJS, but support Jest 27 + asyncTransformers
- next -> switch to ESM

**Comment by bhvngt** [comment]
Yep... After juggling through myriad of configuration I could finally make it work. This was a fun little exercise. Here's the [sandbox](https://codesandbox.io/s/dreamy-water-bcnmd?file=/package.json:392-661) that can be used as a reference for anyone stuck with planning to port svelte, jest and typescript to ESM.

**Comment by sebastianrothe** [comment]
> Yep... After juggling through myriad of configuration I could finally make it work. This was a fun little exercise. Here's the [sandbox](https://codesandbox.io/s/dreamy-water-bcnmd?file=/package.json:392-661) that can be used as a reference for anyone stuck with planning to port svelte, jest and typescript to ESM.

I feel you. And this is supposed to be the future :D

File: src/transformer.js (hunk 1 above)
Comment (sebastianrothe) [reply src/transformer.js:53]: The thing is, there is no documentation about asyncTransformers :( https://jestjs.io/docs/code-transformation#writing-custom-transformers

**Pull request merged**